src/commons/eventsourcing/store.rs

```diff
@@ -178,152 +92,94 @@ where
+    /// or corrupt commands. Will archive any non-snapshot / non-command keys.
     pub fn recover(&self) -> StoreResult<()> {
-        let criteria = CommandHistoryCriteria::default();
+        // TODO: See issue #1086
         for handle in self.list()? {
             info!("Will recover state for '{}'", &handle);
 
-            // Check
-            // - All commands, archive bad commands
-            // - All events, archive bad events
-            // - Keep track of last known good command and event
-            // - Archive all commands and events after
-            //
-            // Rebuild state up to event:
-            //   - use snapshot - archive if bad
-            //   - use back-up snapshot if snapshot is no good - archive if bad
-            //   - start from init event if back-up snapshot is bad, or if the version exceeds last good event
-            //   - process events from (back-up) snapshot up to last good event
-            //
-            //  If still good:
-            //   - save snapshot
-            //   - save info
-
-            let mut last_good_cmd = 0;
-            let mut last_good_evt = 0;
-            let mut last_update = Time::now();
-
-            // Check all commands and associated events
-            let mut all_ok = true;
-
-            let command_keys = self.command_keys_ascending(&handle, &criteria)?;
-            info!("Processing {} commands for {}", command_keys.len(), handle);
-            for (counter, command_key) in command_keys.into_iter().enumerate() {
-                if counter % 100 == 0 {
-                    info!("Processed {} commands", counter);
-                }
+            let scope = Scope::from_segment(Segment::parse_lossy(handle.as_str()));
 
-                if all_ok {
-                    if let Ok(cmd) = self.get_command::<A::StorableCommandDetails>(&handle, &command_key) {
-                        if let Some(events) = cmd.effect().events() {
-                            for version in events {
-                                if let Ok(Some(_)) = self.get_event::<A::Event>(&handle, *version) {
-                                    last_good_evt = *version;
-                                } else {
-                                    all_ok = false;
-                                }
-                            }
+            // If there is not even a valid init command for the
+            // aggregate, then we can really only drop it altogether.
+
+            let aggregate_opt = match self.get_command(&handle, 0) {
+                Err(_) => None,
+                Ok(init_command) => {
+                    if let Some(init_event) = init_command.into_init() {
+                        // Initialise
+                        let mut aggregate = A::init(handle.clone(), init_event);
+
+                        // Find the next command and apply it until there
+                        // is no (valid) next command.
```

Comment:
Did you deliberately drop the progress reporting, is that an accident, or on the TODO list for later?

Reply:
I no longer know the length of the list in advance. And.. we may need to / want to drop this anyway. See issue #1086. In a nutshell: recover is not reliable, it's based on replaying past command/events to a last 'good' state, but that does not work for types that are not event sourced - such as the CaObjects struct.